### PR TITLE
Add analysislogging into make_coinc_search_workflow

### LINKED ISSUE #370: Workflow generators should produce an XML file with a process and process_params table
The workflow generators store their state in the parsed ini files, but for bookkeeping purposes, it will be useful to know exactly how they were invoked. They should create a LIGO_LW XML file with a process and process_params table containing the appropriate metadata. This should also be displayed in the configuration section of the summary pages.

We should definitely do this for the coinc_search and psd_estimation workflows for 1.2.0, but it should be added to the GRB workflow eventually as well.

### PR DESCRIPTION
**Pull request by spxiwh**
This patch adds the analysislogging module into pycbc_make_coinc_search_workflow. I've tested that this generates the appropriate file at runtime. This should resolve #370, although the still outstanding #127 made this more difficult than it should have been. Once v1.2.0 is released we should spend some time on code clean up tasks.

### CODE REVIEW

**Comment by ahnitz** [comment]
Ian, my problem with the segment module and the analysis logging module is the opaque list of segment files. The separate segment list should be visible in the top-level workflow. I'm fine if they are brought together, but that shouldn't happen until calling the function to dump them to disk. 

Take a look at lines 284 -> 299, 

https://github.com/spxiwh/pycbc/blob/add_anal_logging/bin/hdfcoinc/pycbc_make_coinc_search_workflow#L284

Perhaps the output file naming could be put into an overall setup function, as I don't think that's critical, but I think having the input be a list of files which is collected throughout the workflow, and a list of names is preferable. 

Is there a middle ground here, what do you think of the analysis logging module taking over the duty of these lines (except for the file / name collection lists) and also using the the analysislogging modules correct generation of the process params? 

I am OK if we merge this as is, as it satisfies getting the process params into a file, but we need to get back to this and clean up the duplication of code here.

**Comment by spxiwh** [comment]
Hi Alex,

I agree there is code duplication here. I think most of this stems from the different handling of the segment module itself, ie #127. Right now the analysislogging module will not work as intended in the workflow, but it will produce the process_params table, which fulfills the requirement for 1.2.0.

I agree with the proposal of moving lines 284 -> 299 into the analysislogging module (and maybe rename the module), but we should resolve #127 first.

I actually think we should revisit the class currently calles OutSegFile, which is supposed to allow File objects, which are actually segment XML files, to get direct access to the segment list. If we converted that to the glue segment dictionary object, it would allow direct access to all segment lists stored in this, and other, files.

Can we not make that a requirement for v1.2.0 though?

(P.S. You're always welcome to jump on a train and come over here for the day (or more) if there's stuff like this that is easiest to resolve in person).

**Comment by ahnitz** [comment]
Yeah, I agree Ian, we don't need this sorted for 1.2.0, and resolving #127
should be the next highest priority on this front.

I'll have to take up your offer sometime soon, it would be a lot easier to
work out in person I think as there are many interconnected issues, and it
seems difficult to talk about all of them at once.

File: pycbc/workflow/segment.py [hunk 1]
```diff
@@ -1122,9 +1122,15 @@ def get_analyzable_segments(workflow, out_dir, tags=[]):
...
         sci_segs[ifo].coalesce()
-        seg_ok_path = os.path.abspath(os.path.join(out_dir, '%s-SCIENCE-OK.xml' % ifo))
-        seg_files += [segments_to_file(sci_segs[ifo], seg_ok_path, 
-                                          "SCIENCE_OK", ifo=ifo)]  
+        seg_ok_path = os.path.abspath(os.path.join(out_dir,
+                                                   '%s-SCIENCE-OK.xml' % ifo))
+        curr_url = urlparse.urlunparse(['file', 'localhost', seg_ok_path,
+                          None, None, None])
+        curr_file = OutSegFile(ifo, 'SCIENCE_OK', workflow.analysis_time,
+                               curr_url, segment_list=sci_segs[ifo],
```
Comment (ahnitz) [inline pycbc/workflow/segment.py]: Ok, it turns out this change has broken the psd estimation workflow, as the previous function set a PFN automatically, but this does, not. I'm going to temporarily set the PFN here, but I would like to revert to the previous function in a future patch. (but perhaps return an instance of the SegFile class instead)

Comment (spxiwh) [inline pycbc/workflow/segment.py]: Can you set it here:

```
diff --git a/pycbc/workflow/core.py b/pycbc/workflow/core.py
index 1177640..9b7e432 100644
--- a/pycbc/workflow/core.py
+++ b/pycbc/workflow/core.py
@@ -1210,6 +1210,7 @@ class OutSegFile(File):
         from pycbc.events import segments_to_file
         segments_to_file(self.segmentList, self.storage_path, 
                              self.tagged_description,  ifo=self.ifo_string)
+        self.PFN(self.storage_path, site='local')
 
 def make_external_call(cmdList, out_dir=None, out_basename='external_call',
                        shell=False, fail_on_error=True):
```

I can test and commit this myself, unless it is easy for you to do it?

Comment (ahnitz) [inline pycbc/workflow/segment.py]: I've already added it. I needed it to test the psd workflow html page, so
I'll include the commit in the pull request.

**Pull request merged**